drivers/display/display_gc9a01a.h

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright (c) 2017 Jan Van Winkel <jan.van_winkel@dxplore.eu>
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ * Copyright (c) 2020 Teslabs Engineering S.L.
+ * Copyright (c) 2021 Krivorot Oleg <krivorot.oleg@gmail.com>
+ * Copyright (c) 2023 Mr Beam Lasers GmbH.
+ * Copyright (c) 2023 Amrith Venkat Kesavamoorthi <amrith@mr-beam.org> 
+ * 
+ * SPDX-License-Identifier: Apache-2.0
+ * 
+ * @see https://www.buydisplay.com/download/ic/GC9A01A.pdf
+ */
+#ifndef ZEPHYR_DRIVERS_DISPLAY_DISPLAY_GC9A01A_H_
+#define ZEPHYR_DRIVERS_DISPLAY_DISPLAY_GC9A01A_H_
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/sys/util.h>
+
+/* Command registers*/
+// #define GC9A01A_SWRESET 0x01 ///< Software reset register
+// #define GC9A01A 0x04   ///< Read display identification information
+#define GC9A01A 0x09   ///< Read Display Status
+
+#define GC9A01A_SLPIN 0x10  ///< Enter Sleep Mode
+#define GC9A01A_SLPOUT 0x11 ///< Sleep Out
+#define GC9A01A_PTLON 0x12  ///< Partial Mode ON
+#define GC9A01A_NORON 0x13  ///< Normal Display Mode ON
+
+#define GC9A01A_INVOFF 0x20   ///< Display Inversion OFF
+#define GC9A01A_INVON 0x21    ///< Display Inversion ON
+#define GC9A01A_DISPOFF 0x28  ///< Display OFF
+#define GC9A01A_DISPON 0x29   ///< Display ON
+
+#define GC9A01A_CASET 0x2A ///< Column Address Set
+#define GC9A01A_PASET 0x2B ///< Page Address Set
+#define GC9A01A_RAMWR 0x2C ///< Memory Write
+
+#define GC9A01A_PTLAR 0x30    ///< Partial Area
+#define GC9A01A_VSCRDEF 0x33  ///< Vertical Scrolling Definition
+#define GC9A01A_TEOFF 0x34    ///< Tearing effect line off
+#define GC9A01A_TEON 0x35     ///< Tearing effect line on
+#define GC9A01A_MADCTL 0x36   ///< Memory Access Control
+#define GC9A01A_VSCRSADD 0x37 ///< Vertical Scrolling Start Address
+#define GC9A01A_PIXFMT 0x3A   ///< COLMOD: Pixel Format Set
```

Comment:
Usually when writing drivers, especially the first one, it is a good idea to use an existing one as a template. There would be copyright notices from the original or forgotten comments to change, but here `///< COLMOD: Pixel Format Set` looks a bit confusing. Searching for this pattern we find [Adafruit_GC9A01A.h](https://github.com/adafruit/Adafruit_GC9A01A/commit/763dc26bfe95276044fdb1b8c12a08cb0df9e1d9#diff-0e1829dee3da6a0668fb2a86074cd36e018f139b89dd906dfaa1445d87ba20e5)
![adafruit_gc9a01a](https://github.com/zephyrproject-rtos/zephyr/assets/9864546/ef95f4dd-2ff0-42f7-bcb0-26b383e83fca)
This is not acceptable.

Reply:
https://docs.zephyrproject.org/latest/contribute/guidelines.html#licensing It is not worth not having an Apache license for it as you can type these defines in relatively short time from the datasheet.